core/templates/dev/head/services/id-generation.service.spec.ts

```diff
@@ -34,4 +34,10 @@ describe('IdGenerationService', () => {
     let id2 = idGenerationService.generateNewId();
     expect(id1).not.toEqual(id2);
   });
+
+  it('should generate id with 10 digits when random string is less than' +
+    ' 10 of length', function() {
+    spyOn(Number.prototype, 'toString').and.returnValue('0.abcdefhij');
```

Comment:
I think, instead of doing this, you should mock Math.random(). Have a test where it produces a specific small number that leads to less than 10 chars in the string, and a separate test where it produces a number which leads to exactly 10 chars in the string. It seems weird to mock something as fundamental as toString(), but it makes more sense to mock something like random() to make it deterministic for the purposes of tests.

Reply:
Done.